jobserver/api.py

```diff
@@ -114,7 +114,7 @@ def post(self, request, *args, **kwargs):
             log.info(
                 f"About to delete jobs with identifiers: {','.join(identifiers_to_delete)}",
             )
-            job_request.jobs.exclude(identifier__in=identifiers_to_delete).delete()
+            job_request.jobs.filter(identifier__in=identifiers_to_delete).delete()
```

Comment:
Is it worth wrapping this block with `if identifiers_to_delete` so we avoid some empty log lines and no-op database calls?

Reply:
Yes, good point, the last thing we need is red herring logs!